### PR TITLE
[dotnet] Add nullability to Logs API

### PR DESCRIPTION
**Pull request by RenderMichael**
### **User description**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

Adds nullable reference type annotations to logs API.

### Motivation and Context

Contributes to #14640 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

___

### **PR Type**
enhancement, error handling

___

### **Description**
- Enabled nullable reference types across multiple files in the Logs API.
- Added `ArgumentNullException` checks for parameters that should not be null, improving error handling.
- Converted fields to auto-properties with default values in `LogEntry`.
- Improved dictionary parsing and response handling with null checks to ensure robustness.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>ILogs.cs</strong><dd><code>Enable nullability and add argument null check in ILogs</code>&nbsp; &nbsp; </dd></summary>
<hr>

dotnet/src/webdriver/ILogs.cs

<li>Enabled nullable reference types.<br> <li> Added <code>ArgumentNullException</code> for <code>logKind</code> parameter.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14875/files#diff-0954eb8c7c2acd2aade21866339268d7a2220cd675f9fa50b8c7d7e5b2b873d1">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>LogEntry.cs</strong><dd><code>Enable nullability and improve LogEntry parsing</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

dotnet/src/webdriver/LogEntry.cs

<li>Enabled nullable reference types.<br> <li> Converted fields to auto-properties with default values.<br> <li> Improved dictionary parsing with null checks.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14875/files#diff-7a5222bce9dacc30f80638bbbab3ec05e773e6bcd6acab4b27957a47169a9bd9">+19/-28</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>LogLevel.cs</strong><dd><code>Enable nullability in LogLevel</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

dotnet/src/webdriver/LogLevel.cs

- Enabled nullable reference types.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14875/files#diff-a2898a7b00537f5805abbf7173c0aab19d804b87c238d2092cc4aac5dc99a5ff">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>Logs.cs</strong><dd><code>Enable nullability and add argument null checks in Logs</code>&nbsp; &nbsp; </dd></summary>
<hr>

dotnet/src/webdriver/Logs.cs

<li>Enabled nullable reference types.<br> <li> Added <code>ArgumentNullException</code> for <code>driver</code> and <code>logKind</code>.<br> <li> Improved response handling with null checks.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14875/files#diff-0b16524572cd32b7861942211863d45c04da4e9dd6bcd2c4ad0cae47b3fe243a">+16/-10</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: dotnet/src/webdriver/Logs.cs [hunk 1]
```diff
@@ -74,21 +76,25 @@ public ReadOnlyCollection<string> AvailableLogTypes
         /// <param name="logKind">The log for which to retrieve the log entries.
         /// Log types can be found in the <see cref="LogType"/> class.</param>
         /// <returns>The list of <see cref="LogEntry"/> objects for the specified log.</returns>
+        /// <exception cref="ArgumentNullException">If <paramref name="logKind"/> is <see langword="null"/>.</exception>
         public ReadOnlyCollection<LogEntry> GetLog(string logKind)
         {
+            if (logKind is null)
+            {
+                throw new ArgumentNullException(nameof(logKind));
```
Comment (RenderMichael) [inline dotnet/src/webdriver/Logs.cs:84]: The existing behavior for passing in null is exceptional 

image

File: dotnet/src/webdriver/LogEntry.cs [hunk 2]
```diff
@@ -78,32 +70,31 @@ public override string ToString()
...
             }
 
-            if (entryDictionary.ContainsKey("timestamp"))
+            if (entryDictionary.TryGetValue("timestamp", out object? timestamp))
             {
-                DateTime zeroDate = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
-                double timestampValue = Convert.ToDouble(entryDictionary["timestamp"], CultureInfo.InvariantCulture);
-                entry.timestamp = zeroDate.AddMilliseconds(timestampValue);
+                double timestampValue = Convert.ToDouble(timestamp, CultureInfo.InvariantCulture);
+                entry.Timestamp = UnixEpoch.AddMilliseconds(timestampValue);
```
Comment (nvborisenko) [inline dotnet/src/webdriver/LogEntry.cs:84]: `DateTimeOffset.FromUnixTimeMilliseconds()` would be better?

File: dotnet/src/webdriver/Logs.cs (hunk 1 above)
Comment (nvborisenko) [reply dotnet/src/webdriver/Logs.cs:84]: Thx, so null is not allowed.

File: dotnet/src/webdriver/LogEntry.cs (hunk 2 above)
Comment (nvborisenko) [reply dotnet/src/webdriver/LogEntry.cs:84]: No, it is simpler, but not performant.

```
| Method | Mean      | Error     | StdDev    | Allocated |
|------- |----------:|----------:|----------:|----------:|
| Old    | 0.2580 ns | 0.0064 ns | 0.0057 ns |         - |
| New    | 2.3752 ns | 0.0187 ns | 0.0175 ns |         - |
```

Let us keep old approach.

File: dotnet/src/webdriver/LogEntry.cs [hunk 3]
```diff
@@ -42,34 +41,27 @@ private LogEntry()
...
         /// <summary>
         /// Gets the message of the log entry.
         /// </summary>
-        public string Message
-        {
-            get { return this.message; }
-        }
+        public string Message { get; private set; } = string.Empty;
+
+        private static DateTime UnixEpoch => new(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
```
Comment (nvborisenko) [inline dotnet/src/webdriver/LogEntry.cs]: Should be `field`, initialized once.

Comment (RenderMichael) [reply dotnet/src/webdriver/LogEntry.cs]: Done ✅

**Review by nvborisenko — APPROVED** [review]
Thanks @RenderMichael !

**Pull request merged**